planet/src/main/webapp/js/profile-events-renderer.js

```diff
@@ -68,13 +69,18 @@ function getListFromSnapshot(index, snapshot) {
 function getListFromObject(object) {
     let list = [];
     for (const property in object) {
-        object[property].id = property;
-        list.push(object[property]);
+        // Within the list of events, we have the date property
+        if (property !== 'date') {
+            object[property].id = property;
```

Comment:
What is this line doing?

Reply:
It's adding the id of the event to the object (event) which can then be used to reference the event ... The event as it is in the database doesn't have a property id so I needed to save it!

Reply:
What type is |object| supposed to be here?

Reply:
|object| stands for the list of events snapshot so it's an object. I have just changed the names of variables and functions to make more readable!